### PR TITLE
dev-python/elasticsearch: add 8.13.0 + friends and new deps

### PR DESCRIPTION
**Pull request by parona-source**
* Dropped keywords due to new dependencies.

### CODE REVIEW

File: dev-python/respx/respx-0.21.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	dev-python/httpx[${PYTHON_USEDEP}]
+"
+BDEPEND="
+	test? (
+		${RDEPEND}
```
Comment (mgorny) [inline dev-python/respx/respx-0.21.1.ebuild:32]: `det` does that.

File: dev-python/respx/respx-0.21.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		dev-python/httpcore[${PYTHON_USEDEP}]
+		dev-python/starlette[${PYTHON_USEDEP}]
+		dev-python/trio[${PYTHON_USEDEP}]
+	)
+"
+
+distutils_enable_tests pytest
+
+python_test() {
+	epytest -p no:* -o addopts=
```
Comment (mgorny) [inline dev-python/respx/respx-0.21.1.ebuild:43]: Wait, you can do it like this? i.e. don't have to do the whole `PYTEST_DISABLE_PLUGIN_AUTOLOAD` thing? In any case, please quote it, so `*` doesn't accidentally glob something.

File: dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+"
+
+distutils_enable_tests pytest
+
+src_prepare() {
+	default
+
+	# Unnecessary restriction
+	# https://github.com/open-telemetry/opentelemetry-python/pull/3576
+	sed -i -e 's/importlib-metadata >= 6.0, < 7.0/importlib-metadata >= 6.0/' pyproject.toml || die
```
Comment (mgorny) [inline dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild:59]: `/importlib-metadata/s:, < 7.0::`

File: dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=hatchling
+PYTHON_COMPAT=( python3_{11..12} )
+
+OPENTELEMETRY_PUBLIC="1.23.0"
```
Comment (mgorny) [inline dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild:9]: This is… PV?

File: dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+OPENTELEMETRY_PRIVATE="0.44_beta0"
```
Comment (mgorny) [inline dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild:10]: This isn't used anywhere.

File: dev-python/opentelemetry-semantic-conventions/opentelemetry-semantic-conventions-0.44_beta0.ebuild [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=hatchling
+PYTHON_COMPAT=( python3_{11..12} )
+
+OPENTELEMETRY_PUBLIC="1.23.0"
+OPENTELEMETRY_PRIVATE="0.44_beta0"
```
Comment (mgorny) [inline dev-python/opentelemetry-semantic-conventions/opentelemetry-semantic-conventions-0.44_beta0.ebuild:10]: What's with this version? Is it the version installed by the subcomponent?

File: dev-python/elastic-transport/elastic-transport-8.13.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	dev-python/certifi[${PYTHON_USEDEP}]
+	>=dev-python/urllib3-1.26.2[${PYTHON_USEDEP}]
+"
+BDEPEND="
+	test? (
+		dev-python/aiohttp[${PYTHON_USEDEP}]
+		dev-python/httpx[${PYTHON_USEDEP}]
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/opentelemetry-api[${PYTHON_USEDEP}]
+		dev-python/opentelemetry-sdk[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/elastic-transport/elastic-transport-8.13.0.ebuild:39]: Ok, question: does this mean that ES now uses some kind of opt-out telemetry? If yes, then we really want to patch it out and make it opt-in (see `dev-python/stripe`).

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by parona-source** [comment]
In regard to 

```
OPENTELEMETRY_PUBLIC="1.23.0"
OPENTELEMETRY_PRIVATE="0.44_beta0"
```

opentelemetry-python has two versions for its packages a public and a private https://github.com/open-telemetry/opentelemetry-python/releases/tag/v1.23.0

I found it reasonable to include both in all ebuilds so that during an ebuild bump someone couldn't forget about bumping the private version.

Currently the only use of the private version is in relation opentelemetry-semantic-conventions which could alternatively use the public version, but this would require patching metadata.

https://github.com/open-telemetry/opentelemetry-python/blob/v1.23.0/opentelemetry-sdk/pyproject.toml#L29

File: dev-python/respx/respx-0.21.1.ebuild (hunk 1 above)
Comment (parona-source) [reply dev-python/respx/respx-0.21.1.ebuild:32]: I don't know what you mean here.

File: dev-python/respx/respx-0.21.1.ebuild (hunk 2 above)
Comment (parona-source) [reply dev-python/respx/respx-0.21.1.ebuild:43]: In fact setting `PYTEST_DISABLE_PLUGIN_AUTOLOAD` wasn't good enough for this test for whatever reason.

https://github.com/lundberg/respx/blob/master/tests/test_mock.py#L369

File: dev-python/respx/respx-0.21.1.ebuild (hunk 1 above)
Comment (mgorny) [reply dev-python/respx/respx-0.21.1.ebuild:32]: `distutils_enable_tests` does that.

File: dev-python/respx/respx-0.21.1.ebuild (hunk 2 above)
Comment (mgorny) [reply dev-python/respx/respx-0.21.1.ebuild]: Oh, perhaps `-p no:*` also enables "builtin" plugins.

**Comment by mgorny** [comment]
Jeez, what a mess. How about we just use the "public" version as PV, and let the Python package version disagree? It will be a bit confusing, I guess, but still better than having to manually update the version on every single bump.

File: dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+HOMEPAGE="
+	https://opentelemetry.io/
+	https://pypi.org/project/opentelemetry-api/
+	https://github.com/open-telemetry/opentelemetry-python/
+"
+SRC_URI="
+	https://github.com/open-telemetry/opentelemetry-python/archive/refs/tags/v${PV}.tar.gz
+		-> opentelemetry-python-${PV}.gh.tar.gz
+"
+S="${WORKDIR}/opentelemetry-python-${PV}/${PN}"
```
Comment (mgorny) [inline dev-python/opentelemetry-api/opentelemetry-api-1.23.0.ebuild:21]: This one's really shouting for `MY_P="opentelemetry-python-${PV}"`.

File: dev-python/opentelemetry-semantic-conventions/opentelemetry-semantic-conventions-1.23.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+"
+
+distutils_enable_tests pytest
+
+src_prepare() {
+	default
+
+	# Use the same version with all opentelemetry components
+	# https://github.com/gentoo/gentoo/pull/35962#issuecomment-2025466313
+	sed -i -e "s/\(__version__ =\) .*/\1 \"${PV}\"/" src/opentelemetry/semconv/version.py || die
```
[the remark this answers is not in the document]
Comment (mgorny) [reply dev-python/opentelemetry-semantic-conventions/opentelemetry-semantic-conventions-1.23.0.ebuild:55]: To be honest, I think mismatched `.dist-info` version will cause even more confusion.

[the remark this answers is not in the document]
Comment (parona-source) [reply dev-python/opentelemetry-semantic-conventions/opentelemetry-semantic-conventions-1.23.0.ebuild:55]: .dist-info isn't mismatched as version.py is read for this information.

https://github.com/open-telemetry/opentelemetry-python/blob/v1.23.0/opentelemetry-semantic-conventions/pyproject.toml#L34

**Comment by parona-source** [comment]
https://github.com/open-telemetry/opentelemetry-python/releases/tag/v1.24.0

Cool, new update.

**Comment by mgorny** [comment]
Thanks!

**Pull request closed without merging**